src/response/hessian.jl

```diff
@@ -217,15 +322,66 @@ using a split algorithm, where `rhs` is typically
         end
     end
 
-    (; δψ, δoccupation, δεF) = apply_χ0_4P(ham, ψ, occupation, εF, eigenvalues, δHψ;
-                                           occupation_threshold, tol=tol_sternheimer, q,
-                                           kwargs...)
+    # Compute final orbital response
+    # TODO Here we just use what DFTK did before the inexact Krylov business, namely
+    #      a fixed Sternheimer tolerance of tol / 10. There are probably
+    #      smarter things one could do here
+    resfinal = apply_χ0_4P(ham, ψ, occupation, εF, eigenvalues, δHψ;
+                           maxiter=maxiter_sternheimer, tol=tol * factor_final,
+                           bandtolalg, occupation_threshold, q, kwargs...)
+    callback((; stage=:final, runtime_ns=time_ns() - start_ns,
+                Axinfos=[(; basis, tol=tol*factor_final, resfinal...)]))
 
-    (; δψ, δρ, δHψ, δVind, δeigenvalues, δoccupation, δεF, info_gmres)
+    (; resfinal.δψ, δρ, δHψ, δVind, δρ0, δeigenvalues, resfinal.δoccupation,
+       resfinal.δεF, ε, info_gmres)
 end
 
 function solve_ΩplusK_split(scfres::NamedTuple, rhs; kwargs...)
+    if (scfres.mixing isa KerkerMixing || scfres.mixing isa KerkerDosMixing)
+        mixing = scfres.mixing
+    else
+        mixing = SimpleMixing()
+    end
     solve_ΩplusK_split(scfres.ham, scfres.ρ, scfres.ψ, scfres.occupation,
                        scfres.εF, scfres.eigenvalues, rhs;
-                       scfres.occupation_threshold, kwargs...)
+                       scfres.occupation_threshold, mixing,
+                       bandtolalg=BandtolBalanced(scfres), kwargs...)
+end
+
+# TODO This feels a bit hackish, because for all other entities
+#      (like Ω+K or χ0 or K) we don't have a struct, but just an apply function
+#      so why have a struct here ?
+struct DielectricAdjoint{Tρ, Tψ, Toccupation, TεF, Teigenvalues, Tq}
```

Comment:
This is because inexact_mul is with a struct. You could also have inexact_mul be a function (ie gmres(inexact_mul, ...)). I don't have a strong opinion one side or the other.